packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3261,10 +3254,7 @@ export class ContainerRuntime
 						checkpoint.rollback((message: LocalBatchMessage) =>
 							this.rollback(message.runtimeOp, message.localOpMetadata),
 						);
-						// reset the dirty state after rollback to what it was before to keep it consistent
-						if (this.dirtyContainer !== checkpointDirtyState) {
-							this.updateDocumentDirtyState(checkpointDirtyState);
-						}
+						this.updateDocumentDirtyState();
```

Comment:
This is another place where we recompute rather than restoring the previous last-emitted.  Won't affect summarizer (we don't use Order Sequentially there), so should be ok.

Reply:
Fixed - Updated the "dirty" computation to account for these all the time.